### PR TITLE
prevent uninstalling market app from within market app

### LINKED ISSUE #113: Uninstalling market app from market should go to files view and refresh page.
### Steps to reproduce

0. Having a server with market app installed and enabled.
1. Using admin user go to market.
2. Search for market app.
3. Uninstall it.

### Expected behaviour
It shows the message the app has been uninstalled correctly and goes back to the files view, refreshing the page and making unable to keep using market.

### Actual behaviour
No refresh. You can still keep using market seeing things fail.

### Server configuration

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
7.0

**ownCloud version:** (see ownCloud admin page)
current master
{"installed":"true","maintenance":"false","needsDbUpgrade":"false","version":"10.0.2.1","versionstring":"10.0.2.1","edition":"Enterprise","productname":"ownCloud"}

**Updated from an older ownCloud or fresh install:**
Fresh

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No.

**Are you using encryption:** 
No

**Logs**

### Client configuration

**Browser**
Chrome

### PR DESCRIPTION
**Pull request by patrickjahns**
Fixes #113 

We probably should not show the Uninstall button in the frontend

### CODE REVIEW

**Comment by PVince81** [comment]
A better fix would maybe be to add the app here: https://github.com/owncloud/core/blob/master/core/shipped.json#L3

But also need to properly retest 9.x -> 10 upgrades

**Comment by patrickjahns** [comment]
Will we always ship the market app? 

Will provide a PR against core and test it

**Comment by PVince81** [comment]
Let's not merge this and keep this for the next version.

Yeah, let's test with the "shipped.json" entry. I don't know why we didn't add it there. Maybe we forgot or maybe we even removed it on purpose ?

**Comment by PVince81** [comment]
@IljaN please revive and also hide the button in the UI

**Comment by IljaN** [comment]
@PVince81 Done. Backport request is probably outdated?

**Comment by IljaN** [comment]
Adding tests

File: lib/MarketService.php [hunk 1]
```diff
@@ -184,9 +184,10 @@ public function uninstallApp($appId) {
 			throw new \Exception("Installing apps is not supported because the app folder is not writable.");
 		}
 
-		if ($this->appManager->isShipped($appId)) {
+		if ($appId === 'market' || $this->appManager->isShipped($appId)) {
```
Comment (IljaN) [inline lib/MarketService.php]: @PVince81 Does this line make sense if the app is shipped?

Comment (PVince81) [reply lib/MarketService.php]: the app is not part of "core/shipped.json"

and I think that if it was, you wouldn't be able to update it

Comment (IljaN) [reply lib/MarketService.php]: :+1:

Comment (IljaN) [reply lib/MarketService.php]: @PVince81 Does it make sense to add a extra message since "Shipped apps can not be uninstalled" is not true then?

Comment (PVince81) [reply lib/MarketService.php]: fine by me

**Comment by IljaN** [comment]
Ready for review

**Pull request merged**